site/docs/tutorial/sql_to_substrait.md

```diff
@@ -0,0 +1,245 @@
+# SQL to Substrait tutorial
+
+To provide an introduction to Substrait for those already familiar with SQL,
+this tutorial walks through the basics of Substrait by translating basic SQL
+queries. We'll present the Substrait in JSON form so it is (somewhat) readable
+for newcomers.
+
+Substrait can be a type, an expression, a relation, or a plan. We'll look at them in
+those order, going from simplest to most complex.
+
+## Types and Schemas
+
+<!-- TODO -->
+
+## Expressions
+
+A basic building block of queries are expressions. Let's start simple with the
+SQL expression:
+
+```sql
+a > 10
+```
+
+An expression can be one of several things. `a` itself will be translated to
+a field reference:
+
+```json
+{
+  "selection": {
+    "directReference": {
+      "structField": {
+        "field": 1
+      }
+    },
+    "rootReference": {}
+  }
+}
+```
+
+!!! note
+    Protobuf may not serialize fields with integer type and value 0, since 0 is the default.
+    So if you instead saw `"structField": {}`, know that is is equivalent to 
+    `"structField": { "field": 0 }`.
+
+`10` will be translated to a literal expression:
+
+```json
+{
+  "literal": {
+    "i8": 10,
+    "nullable": true
+  }
+}
+```
+
+Then full expression `a > 10` will be a scalar function expression, and will 
+contain the expressions for `a` and `10` in it's `arguments` field:
+
+```json
+{
+  "scalarFunction": {
+    "functionReference": 2,
+    "outputType": {
+      "bool": {
+        "nullability": "NULLABILITY_NULLABLE"
+      }
+    },
+    "arguments": [
+      {
+        "value": {
+          "selection": {
+            "directReference": {
+              "structField": {
+                "field": 1
+              }
+            },
+            "rootReference": {}
+          }
+        }
+      },
+      {
+        "value": {
+          "literal": {
+            "i8": 10,
+            "nullable": true
+          }
+        }
+      }
+    ]
+  }
+}
+```
+
+The name of the function is defined by the `functionReference`, which points to
+the index of the extension function to use. This is serialized separately, which
+you will see later in serialized plans.
+
+```json
+{
+  "extensionFunction": {
+    "extensionUriReference": 1,
+    "functionAnchor": 2,
+    "name": "gt"
+  }
+}
+```
+
+## Relations
+
+An logical or physical plan is typically represented as a tree of nodes, such
```

Comment:
```suggestion
A logical or physical plan is typically represented as a tree of nodes, such
```
Do we want to introduce or mention "logical" and "physical"?  Won't these terms confuse a reader at this point?

Reply:
It's not clear here, but I was speaking of plans in general, not Substrait plans specifically. I assume that anyone coming from the SQL world has seen some form of logical / physical plan (at the very least, has run `EXPLAIN ...` at some point and tried to read the output). I'll make this more specific so that's clear.